src/systems/input.rs

```diff
@@ -9,17 +10,55 @@ use bevy::prelude::*;
 // Player 2 -> UP -> 'W' Key
 // Player 2 -> DOWN -> 'S' Key
 pub fn keyboard_input_system(
-    mut players: Query<(&mut Transform, &Player)>,
+    mut commands: Commands,
+    mut players_query: Query<(&mut Transform, Entity, Option<&mut InputTimer>, &Player)>,
+    time: Res<Time>,
     key: Res<Input<KeyCode>>,
     window: Res<WindowDescriptor>,
 ) {
-    for (mut transform, player) in players.iter_mut() {
+    for (mut transform, entity, input_timer, player) in players_query.iter_mut() {
         match player.player_type {
             PlayerType::Left => {
                 if key.pressed(KeyCode::W) {
```

Comment:
Could you do something like this to reduce code duplication?
```rust
let t = match input_timer {
    Some(t) => {
        t.tick(time.delta());
        t
    },
    None => {
        let t = timer.clone();
        commands.entity(entity).insert(InputTimer {timer: t});
        t
    }
}

let elapsed = timer.elapsed_secs();
let d_v = get_translation(elapsed);
let v = d_v * PADDLE_VELOCITY;
transform.translation.y += v
```

Reply:
You know I tried, but returning `t` in the `Some(t)` arm is a uses `move` semantics and man that's a fight with the compiler. I will try again.

Reply:
It ended up looking like this

```rust
        let t = match input_timer {
            Some(mut input_timer) => {
                input_timer.timer.tick(time.delta());
                let t = InputTimer {
                    timer: input_timer.timer.clone(),
                };
                t
            }
            None => {
                let timer = Timer::new(Duration::from_secs_f32(1.0), false);
                let t = InputTimer { timer };
                t
            }
        };
```

Big time fighting the compiler for this one to work

Reply:
At least there's less code 😆

```
let t = InputTimer {
    timer: input_timer.timer.clone(),
};
```

One thing I dislike about Bevy is that it forces this kind of weirdness. It should just be `input_timer.close()`  🤷‍♂️

Reply:
It's not bevy, it's rust. I ended up implementing the a clone trait myself on `InputTimer`